src/3d/framegraph/qgsframegraph.h

```diff
@@ -239,34 +202,43 @@ class QgsFrameGraph : public Qt3DCore::QEntity
     */
     void removeClipPlanes();
 
+    /**
+     * Registers a new the render view \a renderView with name \a name
+     *
+     * Will take ownership of the renderView
+     */
+    bool registerRenderView( std::shared_ptr<QgsAbstractRenderView> renderView, const QString &name );
+
+    //! Unregisters the render view named \a name, if any
+    void unregisterRenderView( const QString &name );
+
+    //! Enables or disables the render view named \a name according to \a enable
+    void setRenderViewEnabled( const QString &name, bool enable );
+
+    //! Returns true if the render view named \a name is found and enabled
+    bool isRenderViewEnabled( const QString &name );
+
+    //! Returns shadow renderview or nullptr if not defined
+    QgsShadowRenderView &shadowRenderView() const;
+
+    //! Returns forward renderview or nullptr if not defined
+    QgsForwardRenderView &forwardRenderView() const;
+
+    //! Returns the render view named \a name, if any
+    QgsAbstractRenderView &renderView( const QString &name );
```

Comment:
What if a render view with the given name does not exist? that would return invalid reference which should never happen - the right approach is to return a pointer (or we would need to keep some kind of "null" render view to be able to return a valid reference